app/models/work.rb

```diff
@@ -0,0 +1,33 @@
+class Work < ApplicationRecord
+
+  has_many :votes
+  has_many :users, through: :votes
+
+  validates :title, presence: true
+
+  validates :category, presence: true
+
+  validates :publication_year, presence: true,
+             numericality: { greater_than: 1800, less_than:  Date.current.year + 1 }
+
+
+  def self.albums
+    where(category: "album")
+  end
+
+  def self.books
+    where(category: "book")
+  end
+
+  def self.movies
+    where(category: "movie")
+  end
+
+  def self.media_spotlight
+    Work.all.sort_by { |work| work.votes.count }.reverse[0]
+  end
+
+  def self.top_ten(category)
+    Work.where(category: category).sort_by { |work| work.votes.count }.reverse[0..9]
+  end
```

Comment:
These function, but they're not the most efficient way to do it.  It's better to use the database, through `.order` to sort the data at the database level.  That saves Postgres from having to transmit the information to Rails and Postgres is _really_ good at handling large amounts of data, so it scales well.

You can read more about this in [this blog post](https://github.com/Ada-C14/media-ranker/pull/26).